Src/WitsmlExplorer.Api/Services/LogDataReader.cs

```diff
@@ -103,5 +121,35 @@ public async Task<WitsmlLogData> GetNextBatch()
             _dropFirstRow = true;
             return sourceLogData;
         }
+
+        /// <summary>
+        /// Fetches and returns the next batch of log data. Will throw if an exception has happened during fetching, or on timeout.
+        /// </summary>
+        /// <returns>WitsmlLogData or, if there is no more data to fetch, null</returns>
+        public async Task<WitsmlLogData> GetNextBatch()
+        {
+            if (_exception != null)
+            {
+                throw _exception;
+            }
+            if (_finished && _buffer.Count == 0)
+            {
+                return null;
+            }
+            return await WithTimeout(_buffer.ReceiveAsync());
+        }
+
+        private async Task<T> WithTimeout<T>(Task<T> task)
```

Comment:
I think this method can be replaced by using a `cancellationToken` with a timeout.

Create a field:
` private readonly CancellationTokenSource _cancellationTokenSource;`

In the constructor:
```
const int timeout = 5 * 60 * 1000;
_cancellationTokenSource = new CancellationTokenSource(timeout);
```

In the Produce method:
`await _buffer.SendAsync(logData, _cancellationTokenSource.Token);`

In the GetNextBatch method:
`return await _buffer.ReceiveAsync(_cancellationTokenSource.Token);`

This should handle the timeout in a similar way, while also providing a cancellation token if there are other needs.

Reply:
Rewritten to use cancellationTokenSource and IAsyncDisposable instead of timeout.